src/all/tachidesk/src/eu/kanade/tachiyomi/extension/all/tachidesk/Tachidesk.kt

```diff
@@ -0,0 +1,190 @@
+        return List(chapter.pageCount) {
+            Page(it + 1, "", "$baseUrl/api/v1/manga/$mangaId/chapter/$chapterIndex/page/$it/")
+        }
+    }
+
+    // ------------- Preferences -------------
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        screen.addPreference(screen.editTextPreference(ADDRESS_TITLE, ADDRESS_DEFAULT, baseUrl))
+    }
+
+    /** boilerplate for [EditTextPreference] */
+    private fun PreferenceScreen.editTextPreference(title: String, default: String, value: String, isPassword: Boolean = false): EditTextPreference {
+        return EditTextPreference(context).apply {
+            key = title
+            this.title = title
+            summary = value
+            this.setDefaultValue(default)
+            dialogTitle = title
+
+            if (isPassword) {
+                setOnBindEditTextListener {
+                    it.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+                }
+            }
+
+            setOnPreferenceChangeListener { _, newValue ->
+                try {
+                    val res = preferences.edit().putString(title, newValue as String).commit()
+                    Toast.makeText(context, "Restart Tachiyomi to apply new setting.", Toast.LENGTH_LONG).show()
+                    res
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                    false
+                }
+            }
+        }
+    }
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private fun getPrefBaseUrl(): String = preferences.getString(ADDRESS_TITLE, ADDRESS_DEFAULT)!!
+
+    companion object {
+        private const val ADDRESS_TITLE = "Address"
+        private const val ADDRESS_DEFAULT = ""
+    }
+
+    // ------------- Not Used -------------
+
+    override fun latestUpdatesRequest(page: Int): Request = throw Exception("Not used")
+
+    override fun latestUpdatesParse(response: Response): MangasPage = throw Exception("Not used")
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) = throw Exception("Not used")
+
+    override fun searchMangaParse(response: Response): MangasPage = throw Exception("Not used")
+
+    override fun chapterListParse(response: Response): List<SChapter> = throw Exception("Not used")
+
+    override fun pageListParse(response: Response): List<Page> = throw Exception("Not used")
+
+    override fun imageUrlParse(response: Response): String = throw Exception("Not used")
+
+    // ------------- Util -------------
+
+    private fun MangaDataClass.toSManga() = SManga.create().also {
+        it.title = title
+        it.url = id.toString()
+        it.thumbnail_url = "$baseUrl$thumbnailUrl"
+        it.artist = artist
+        it.author = author
+        it.description = description
+        it.status = when (status) {
+            "UNKNOWN" -> 0
```

Comment:
```suggestion
            "UNKNOWN" -> SManga.UNKNOWN
```

Same with the other ones below this line.

Reply:
Tachidesk supports the same exact 4 possible status, else can only represent an impossible error

Reply:
Using `SManga` constants makes sense, I'll change that bit.

Reply:
I think what he means is remove the 0 branch.  And make it just else at the end. So if tachidesk ever adds different statuses it will fall back to unknown besides going to else when 0

Reply:
I another one is ever added, it'll be mirroring the same thing in Tachiyomi

Reply:
Defensive programming. The manga shouldn't fail to load, just because it doesn't know the status.  Just because you plan at this time to only match Tachiyomi status, doesn't mean someone else who forks tachi desk will, or you won't change your mind down the line.  There is no reason to throw an exception here.

Reply:
alright, makes sense